docs/source/named_tensor.rst

```diff
@@ -0,0 +1,302 @@
+.. currentmodule:: torch
+
+.. _named_tensors-doc:
+
+Named Tensors
+=============
+
+Named Tensors aim to make tensors easier to use by allowing users to associate
+explicit names with tensor dimensions. In most cases, operations that take
+dimension parameters will accept dimension names, avoiding the need to track
+dimensions by index. In addition, named tensors use names to automatically
+check that APIs are being used correctly at runtime, providing extra safety.
+Names can also be used to rearrange dimensions and to align tensors by their
```

Comment:
you haven't defined what "align tensors by their position names mean.  Maybe something like:
"Names can also be used to rearrange dimensions, for example to support "broadcasting by name" rather than "broadcasting by position".